### ISSUE TITLE
Integration with bittorrent

### ISSUE THREAD

**Issue by mrd0ll4r** [nukeop/nuclear#123]
...to truly be popcorntime for music.

I imagine integration with popular private trackers, to avoid downloading any `famous-song.mp3.exe.zip`s. Also they usually have a relatively well-organized library to tap.

**Comment by nukeop** [comment]
I thought about this, but the problem is that most music on torrent trackers is shared via zipped albums, and there's no way to selectively stream a single song. Private trackers would be a good choice but that would require a lot of manual fiddling by the users to use their credentials, and I imagine it would increase the risk of being banned and losing a precious account (or at the very least lowering seed ratio).

**Comment by mrd0ll4r** [comment]
Yes, I'd avoid public trackers, for reasons of malware and the archives issue. I agree working with private trackers might be a bit annoying - you really want to keep the stuff seeding. I still think it would be fantastic, and honestly sometimes ratio doesn't matter as much. As a first approach, all the downloaded data could just be saved locally to at least avoid re-downloading it. Seeding would be nice, but I see how shipping a complete bittorrent client that can handle possibly hundreds of torrents is a bit overkill.

Alternative idea: What about a plugin or any integration of using a library via sftp? As I see it, you're already considering using a local library, so that would probably not be too far off. From there, it would then only be a small-ish step to integrate the whole thing with a remote rtorrent/deluge/... client to handle the actual torrent stuff.

It's all just brainstorming though... But it would be super cool :)

I actually many times thought about a project to unify the APIs of private tracker sites to make the whole integration easier, because I was (and am) very interested in the "popcorntime for music". Maybe something like that would be useful now...

**Comment by JohnDoee** [comment]
If you ever want to add **streaming** support for private trackers, your only torrent client choice is deluge with a plugin I made called [deluge-streaming](https://github.com/JohnDoee/deluge-streaming). 

With music being small files, it might be possible to just download a file before playing it.

> ... unify the APIs of private tracker sites

This can be quite hard as sites have a tendency to spit out data organized differently with very different models related to metadata :smile:

**Comment by Korb** [comment]
> the problem is that most music on torrent trackers is shared via zipped albums

The largest Russian-language torrent tracker directly prohibits posting music in archives. I rarely did not find the discography of some musical artist on it, and I recommend this site for experiments on introducing p2p support into Nuclear Player.

**Comment by nukeop** [comment]
Do you mean rutracker? That would be pretty cool to have.

**Comment by nukeop** [comment]
This issue is from an older version of Nuclear that has been replaced with a complete rewrite. None of the old issues apply to the new codebase.

The old code is preserved on the [legacy/electron](https://github.com/nukeop/nuclear/tree/legacy/electron) branch. If you have feedback about the new version, head to [Discussions](https://github.com/nukeop/nuclear/discussions).
